storage/store.go

```diff
@@ -489,23 +493,97 @@ func (s *Store) Start(stopper *util.Stopper) error {
 // startGossip runs an infinite loop in a goroutine which regularly checks
 // whether the store has a first range replica and asks that range to gossip
 // cluster ID and first range metadata accordingly.
-func (s *Store) startGossip() {
+func (s *Store) startGossip() error {
+	// Take care of gossip for config descriptors. It does not
+	// expire, so this is a one-time action in a working cluster - if values change, ranges
+	// will update gossip autonomously.
+	// However, the lease holder, who is normally in charge of that
+	// might crash precisely when trying to gossip. To account for
+	// this rare scenario, we activate all ranges that hold config
+	// maps periodically.
+	configFn := func() error {
```

Comment:
I'd break this sucker out into an actual method on store. `maybeGossipConfigs() error` or similar.